src/WCSimLIGen.cc

```diff
@@ -0,0 +1,282 @@
+#include "WCSimLIGen.hh"
+#include "WCSimPrimaryGeneratorMessenger.hh"
+#include "WCSimPrimaryGeneratorAction.hh"
+
+#include "json.hpp"
+
+#include "G4ParticleGun.hh"
+#include "G4PhysicalConstants.hh"
+#include "G4OpticalPhoton.hh"
+#include "G4RandomDirection.hh"
+#include "G4SystemOfUnits.hh"
+
+using namespace std;
+using json = nlohmann::json;
+
+WCSimLIGen::WCSimLIGen(){
+
+    // Initialise
+    //myDetector = myDC;
+    //myMessenger = new WCSimLIGenMessenger(this);
+    this->Initialise();
+
+}
+
+WCSimLIGen::~WCSimLIGen(){
+
+    // things to delete?
+    //this->Finalise();
+}
+
+
+void WCSimLIGen::Initialise(){
+
+    // myMessenger, myDetector
+    photonMode = 0;
+    wcsimdir = string(getenv("WCSIMDIR"))+"/data/";
+} 
+
+void WCSimLIGen::SetPhotonMode(G4bool photonmode){
+   
+    if (photonmode){
+        photonMode=1;
+    }
+}
+    
+
+void WCSimLIGen::ReadFromDatabase(G4String injectorType, G4String injectorIdx, G4String injectorFilename){
+
+    // Define the database to read from
+    string db = injectorFilename;
+    if (db.empty()) {
+        db = wcsimdir + "LightInjectors.json";
+    }
+    else {
+        db = wcsimdir + injectorFilename;
+    }
+    
+    ifstream fJson(db.c_str());
+    
+    if (!fJson) {
+        G4cout << "LIGen: [ERROR] light injector db " << db << " not found, reverting to generic db" << G4endl;
+    } else {
+        G4cout << " LIGen: Light Injector db " << db << " opened" << G4endl;
+    }
+    stringstream buffer;
+    buffer << fJson.rdbuf();
+
+    // Read in position and direction of injector from database
+    // and fill the histogram with the light injector profile
+    json data = json::parse(buffer.str());
+    for (auto injector : data["injectors"]){
+        if ( injector["type"].get<string>()==injectorType && G4String(injector["idx"].get<string>())==injectorIdx ){
+
+            G4cout << "Simulating events from " << injector["type"].get<string>() << " " << injector["idx"].get<string>() << G4endl;
+            injectorWavelength = injector["wavelength"].get<double>();
+            injectorPosition = injector["position"].get<vector<double>>();
+            injectorDirection = injector["direction"].get<vector<double>>();
+            injectorOffset = injector["offset"].get<double>();
+            thetaVals = injector["theta"].get<vector<double>>();
+            phiVals = injector["phi"].get<vector<double>>();
+            intensity = injector["intensity"].get<vector<double>>(); 
+            if ( photonMode ){
+                photonsFilename = injector["photonsfile"].get<string>();
+            }
+        }
+    
+    }	
+    
+    if ( photonMode ){
+        LoadPhotonList();
+    }
+    else {
+        LoadProfilePDF();
+    }
+}
+
+void WCSimLIGen::LoadPhotonList(){
+
+    string photonsFile = wcsimdir + photonsFilename;
+    cout << photonsFile << endl; 
```

Comment:
Please use `G4cout` `G4err` and `G4endl` exclusively (there can be some weird mangling of printout if these are mixed with `std::cout` etc)